### DISCUSSION TITLE
concourse in a box: concourse + s3 + vault in docker compose

### DISCUSSION THREAD

**Discussion by marco-m** [concourse/concourse#7018] [Show and tell]
Hello,

Just wanted to share [marco-m/concourse-in-a-box](https://github.com/marco-m/concourse-in-a-box), a simple all-in-one Concourse system based on Docker Compose, with Minio S3-compatible storage and HashiCorp Vault secret manager. This enables to:

* Learn Concourse pipelines from scratch in a simple environment.
* Troubleshoot production Concourse pipelines in a stand-alone environment.
* Write Concourse pipelines that can be reused as-is in your production environment, since it comes with S3 and secret store.

**Comment by taylorsilva** [comment]
So cool! I may try writing tutorials using this in the future.

**Reply by marco-m** [reply]
That would be a great idea indeed!

**Comment by marco-m** [comment]
Hello, updated to Concourse 7.4.0

**Reply by marco-m** [reply]
Added example (with screenshots) about instanced pipelines (https://concourse-ci.org/instanced-pipelines.html)
